File: frontend/src/app/services/models/categories/actions/lobby/lobby-client-action/lobby-client-action-envelope.ts
import {LobbyClientAction} from './lobby-client-action';
import {WebsocketEnvelope} from '../../../../websocket-envelope';

interface LobbyClientActionEnvelope extends WebsocketEnvelope {
  category: 'LOBBY_CLIENT_ACTION';
  payload: LobbyClientAction;
}

export function lobbyClientActionEnvelope(payload: LobbyClientAction): LobbyClientActionEnvelope {
  return {category: 'LOBBY_CLIENT_ACTION', payload};
}

File: backend/src/main/java/dk/mathiaskofod/services/session/envelopes/LobbyClientActionEnvelope.java
package dk.mathiaskofod.services.session.envelopes;

import dk.mathiaskofod.services.session.actions.lobby.client.LobbyClientAction;
import dk.mathiaskofod.services.session.models.annotations.Category;

@Category("LOBBY_CLIENT_ACTION")
public record LobbyClientActionEnvelope(LobbyClientAction payload) implements WebsocketEnvelope<LobbyClientAction> {}
